File: DevExpress-Demo/VB/TreeListDemo.Wpf/Data/FileSystemImages.vb
Imports System.Collections.Generic
Imports System.Windows.Media

Namespace TreeListDemo.Data

    Public Class FileSystemImages

        Private Shared ReadOnly images As Dictionary(Of String, ImageSource) = New Dictionary(Of String, ImageSource)()

        Private Shared Function ImageGet(ByVal name As String) As ImageSource
            Dim image As ImageSource = Nothing
            If images.TryGetValue(name, image) Then Return image
            image = GetSvgImage(name)
            images.Add(name, image)
            Return image
        End Function

        Public Shared ReadOnly Property FileImage As ImageSource
            Get
                Return ImageGet("File")
            End Get
        End Property

        Public Shared ReadOnly Property DiskImage As ImageSource
            Get
                Return ImageGet("Local_Disk")
            End Get
        End Property

        Public Shared ReadOnly Property ClosedFolderImage As ImageSource
            Get
                Return ImageGet("Folder_Closed")
            End Get
        End Property

        Public Shared ReadOnly Property OpenedFolderImage As ImageSource
            Get
                Return ImageGet("Folder_Opened")
            End Get
        End Property
    End Class
End Namespace

File: DevExpress-Demo/CS/TreeListDemo.Wpf/Data/FileSystemImages.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Media;

namespace TreeListDemo.Data {
    public class FileSystemImages {
        private static readonly Dictionary<string, ImageSource> images = new Dictionary<string, ImageSource>();

        private static ImageSource ImageGet(string name) {
            ImageSource image = null;
            if(images.TryGetValue(name, out image))
                return image;
            image = ImageHelper.GetSvgImage(name);
            images.Add(name, image);
            return image;
        }

        public static ImageSource FileImage {
            get {
                return ImageGet("File");
            }
        }
        public static ImageSource DiskImage {
            get {
                return ImageGet("Local_Disk");
            }
        }

        public static ImageSource ClosedFolderImage {
            get {
                return ImageGet("Folder_Closed");
            }
        }
        public static ImageSource OpenedFolderImage {
            get {
                return ImageGet("Folder_Opened");
            }
        }
    }
}
